treeherder/services/pulse/sources.py

```diff
@@ -0,0 +1,40 @@
+import environ
+
+env = environ.Env()
+
+
+def get_sources():
+    """
+    Get ingestion source locations.
+
+    Specifies the Pulse exchanges Treeherder will ingest data from.  This list
+    will be updated as new applications come online that Treeherder supports.
+    Treeherder will subscribe with routing keys that are all combinations of
+    ``project`` and ``destination`` in the form of: <destination>.<project>
+    Wildcards such as ``#`` and ``*`` are supported for either field.
+    """
+    sources = env.json(
+        "PULSE_DATA_INGESTION_SOURCES",
```

Comment:
I'd like to eventually convert this into separate settings, such as:
* exchanges
* projects
* destinations

...with appropriate defaults set for each.

This would mean the config not being quite as configurable (since it would no longer be possible to set a different `projects` filter for each exchange) - however I don't think we ever need anything that complex.

In return, we'd get:
* easier to understand/set config variables (no hard to read JSON blobs in the environment)
* the ability to easily add a project filter without having to copy-paste the list of exchanges

We should also stop setting these environment variables in Heroku, and use the defaults here instead (I don't like hiding settings in Heroku config when they aren't secrets). The environment variables would then only be used in development for testing/filtering etc.

See also [bug 1281821](https://bugzilla.mozilla.org/show_bug.cgi?id=1281821).

Reply:
Agreed, I'll follow up in the bug.